File: 卡码/代码随想录/1/09-动态规划/Js/05-子序列问题/02-子序列(连续)/LeetCode_0718.js
//718. 最长重复子数组
const findLength = function(nums1, nums2) {
	const dp=new Array(nums1.length+1).fill(0).map(item=>new Array(nums2.length+1).fill(0));
	let res=0;
	for(let i=1;i<nums1.length+1;i++){
		for(let j=1;j<nums2.length+1;j++){
			if(nums1[i-1]==nums2[j-1]) dp[i][j]=dp[i-1][j-1]+1;
			res=dp[i][j]>res?dp[i][j]:res;
		}
	}
	return res;
};

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/动态规划/_05_子序列问题/_02_连续子序列/LeetCode_0718.java
package com.kurumi.leetCode.codeThoughts.动态规划._05_子序列问题._02_连续子序列;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 718. 最长重复子数组
 */
public class LeetCode_0718 {
    static class Solution {
        public int findLength(int[] nums1, int[] nums2) {
            int[][] dp = new int[nums1.length + 1][nums2.length + 1];
            int res = 0;
            for (int i = 1; i <= nums1.length; i++) {
                for (int j = 1; j <= nums2.length; j++) {
                    if (nums1[i - 1] == nums2[j - 1])
                        dp[i][j] = dp[i - 1][j - 1] + 1;
                    res = dp[i][j] > res ? dp[i][j] : res;
                }
            }
            return res;
        }
    }
}
